common/src/main/java/org/opensearch/sdk/DeleteDataObjectResponse.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import org.opensearch.action.support.replication.ReplicationResponse.ShardInfo;
+import org.opensearch.core.common.Strings;
+import org.opensearch.core.index.shard.ShardId;
+
+public class DeleteDataObjectResponse {
+    private final String id;
+    private final ShardId shardId;
+    private final ShardInfo shardInfo;
```

Comment:
Why are these fields required? Shard details are specific to opensearch which will make it difficult to extend the interface to non-opensearch data stores.

Reply:
I previously answered that question when you asked it [here](https://github.com/opensearch-project/ml-commons/pull/2430#discussion_r1603931384).

The return value to the end user is created from a `DeleteResponse` object, which _requires_ a non-null ShardId:
```java
this.shardId = Objects.requireNonNull(shardId);
```
So I have to have a Shard ID. I can create one, or I can pass along the actual one that is returned from cluster-based OpenSearch.   Which do you prefer?   I see no good reason to delete information that exists.  It can be ignored for other implementations.

Similarly, for `ShardInfo`, the return value from the Delete Connector API looks like this:
```json
{
    "_index": ".plugins-ml-connector",
    "_id": "qU00aowBH9rhI2IwnrH3",
    "_version": 2,
    "result": "deleted",
    "_shards": {
        "total": 1,
        "successful": 1,
        "failed": 0
    },
    "_seq_no": 2,
    "_primary_term": 1
}
```

I can either pass along those actual statistics or make them up.  Which is easier to explain to a customer, that a statistic might be irrelevant for their storage option, or that we're taking away information they used to have?